ui/lib/core/addon/utils/search-select-has-many.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Util to add/remove models in a hasMany relationship via the search-select component
+ *
+ * @example of using the util within an action in a component
+ *```js
+ * @action
+ * async onSearchSelectChange(selectedIds) {
+ *    const methods = await this.args.model.mfa_methods;
+ *   handleHasManySelection(selectedIds, methods, this.store, 'mfa-method');
+ *  }
+ *
+ * @param selectedIds array of selected options from search-select component
```

Comment:
When you say selected options, does that mean an option that was selected, or is this an array of options they can select?

Reply:
an array of options they selected - this handled by the search-select component:
 `selectedOptions` is returned via `this.onChange` which is sent to the callback function 

```  handleChange() {
    if (this.selectedOptions.length && typeof this.selectedOptions.firstObject === 'object') {
      if (this.passObject) {
        this.onChange(Array.from(this.selectedOptions, (option) => ({ id: option.id, isNew: !!option.new })));
      } else {
        this.onChange(Array.from(this.selectedOptions, (option) => option.id));
      }
    } else {
      this.onChange(this.selectedOptions);
    }
  },

Reply:
Got it. In most cases is selectedOptions one thing? I guess what confuses me, and it's not your language but the original of SearchSelect is why options is plural if this is coming from a search component. I would think you'd be searching, adding just one thing. That's how i've seen the searchSelect component work. Again, not something for you to address, more just the plural is messing with my thinking of what this param represents.

Reply:
The `SearchSelect` component always returns an array in the `onChange` callback even if you can only select a single item. I think for oidc assignments as an example, since entities and groups are arrays you should be able to select multiple via `SearchSelect`.